nanoFramework.Tools.DebugLibrary.Shared/PortComposite/PortCompositeDeviceManager.cs

```diff
@@ -52,21 +49,42 @@ private void OnLogMessageAvailable(object sender, StringEventArgs e)
 
         private void OnPortDeviceEnumerationCompleted(object sender, EventArgs e)
         {
-            DeviceEnumerationCompleted?.Invoke(this, EventArgs.Empty);
+            IsDevicesEnumerationComplete = (from p in _ports
+                                            where p.IsDevicesEnumerationComplete
+                                            select p).Any();
+            if (IsDevicesEnumerationComplete)
+            {
+                DeviceEnumerationCompleted?.Invoke(this, EventArgs.Empty);
+            }
         }
 
         /// <inheritdoc/>
         /// <exception cref="NotImplementedException">This API is not available in </exception>
         public override void AddDevice(string deviceId)
         {
-            _ports.ForEach(p =>
-            {
-                p.AddDevice(deviceId);
-            });
+            // None of the Port*Manager has a check whether deviceId matches the ID handled by the manager.
+            throw new NotImplementedException();
+            //_ports.ForEach(p =>
+            //{
+            //    p.AddDevice(deviceId);
+            //});
+        }
+
+        /// <inheritdoc/>
+        /// <exception cref="NotImplementedException">This API is not available in </exception>
+        public override NanoDeviceBase AddAndReturnDevice(string deviceId)
```

Comment:
I now understand the need to return the device. No need to have this.

(Preferable to actually remove the commented-out code when it's not needed)

Reply:
I don't know whether it is not needed, I just concluded it would not work as expected. :-)